### PR TITLE
(GH-93) Draft Anatomy of a Tool concept document

### LINKED ISSUE #93: Concept: Anatomy of a Tool
## Acceptance Criteria

- [x] Explains the tool config
- [x] Explains the content folder
- [x] Explains the entrypoint script(s)

### PR DESCRIPTION
**Pull request by sanfrancrisko**
Resolves #93.

### CODE REVIEW

File: docs/md/content/usage/tool-authoring.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+category: narrative
+tags:
+  - templates
+  - usage
+weight: 20
+---
+
+## Structure
+
+A PRM tool package can be packaged as a `.tar.gz` or hosted in a Github repository.
```
Comment (da-ar) [inline docs/md/content/usage/tool-authoring.md:13]:
```suggestion
A PRM tool can be packaged as a `.tar.gz` or hosted in a Github repository.
```

File: docs/md/content/usage/tool-authoring.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+It is comprised of a `prm-config.yml` and an optional `content` directory.
```
Comment (da-ar) [inline docs/md/content/usage/tool-authoring.md:14]:
```suggestion
It is comprised of a `prm-config.yml` and an `content` directory.
```
`content` dir is no longer optional

File: docs/md/content/usage/tool-authoring.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+The `content` directory can contain scripts / binaries / config files, etc... required to supplement the functionality of the tool outside the configurable parameters within the `prm-config.yml`.
```
Comment (da-ar) [inline docs/md/content/usage/tool-authoring.md:15]: Add that the contents of this directory are placed in the /tmp directory. So 'content/myfile.sh' will be acessible at `/tmp/myfile.sh`

File: docs/md/content/usage/tool-authoring.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+`env`: A list of environment variables and the value to set them to
+
+<!-- Uncomment when this is implemented -->
+<!-- `output_mode`: The output mode (`junit|json|yaml`) -->
+
+`requires_git`: Set to `true` if this tool requires `git`
+
+`use_script`: The name of the script to invoke to execute the tool.
+This should be present in the `content` dir and in either `.sh` or `.ps1` format.
```
Comment (da-ar) [inline docs/md/content/usage/tool-authoring.md:97]: implementation dependant but this will in `.sh` and `.ps1` format

Comment (da-ar) [reply docs/md/content/usage/tool-authoring.md:97]: also say that `use_script` should withhold the extension i.e `runme.sh` should be defined `use_script: "runme"`

**Review by da-ar — COMMENTED** [review]
some additional context

**Comment by sanfrancrisko** [comment]
Also need to document:
```yaml
puppet: 
  enabled: true
```

File: docs/md/content/usage/tool-authoring.md (hunk 3 above)
Comment (sanfrancrisko) [reply docs/md/content/usage/tool-authoring.md:15]: I'll add something to cover that, although I am trying to strike a balance between being specific enough for the first iteration (Docker backend) and keeping it generic enough so a complete rewrite isn't required when we implement other backends.

File: docs/md/content/usage/tool-authoring.md (hunk 4 above)
Comment (sanfrancrisko) [reply docs/md/content/usage/tool-authoring.md:97]: I might drop the reference to the `.sh|ps1` format for now, on second thoughts - especially given my [prior comment](https://github.com/puppetlabs/prm/pull/72/files#r772198442)

File: docs/md/content/usage/tool-authoring.md (hunk 2 above)
Comment (sanfrancrisko) [reply docs/md/content/usage/tool-authoring.md]: This doesn't seem to be the case:

- I can still omit a `content` dir and it builds OK
- Including an empty `content` dir causes `docker build` to fail

**Pull request merged**